rcl/src/rcl/init.c

```diff
@@ -142,15 +140,17 @@ rcl_init(
 #endif // RCL_COMMAND_LINE_ENABLED
 
   // Set the instance id.
-  uint64_t next_instance_id = rcutils_atomic_fetch_add_uint64_t(&__rcl_next_unique_id, 1);
+  static uint32_t next_instance_id = 0;
+  next_instance_id++;
   if (0 == next_instance_id) {
-    // Roll over occurred, this is an extremely unlikely occurrence.
-    RCL_SET_ERROR_MSG("unique rcl instance ids exhausted");
-    // Roll back to try to avoid the next call succeeding, but there's a data race here.
-    rcutils_atomic_store(&__rcl_next_unique_id, -1);
-    goto fail;
+    next_instance_id++;
+    // // Roll over occurred, this is an extremely unlikely occurrence.
```

Comment:
Should we keep this rollover error check?

Reply:
No, because this fails when there is a rollover that in this RCL instance we cannot determine if instance id 1 is being used. Imagine the situation where 2^64 RCL instances are being run simultaneously.

In our case, only one RCL instance is going to be run at the same time, so we do not care about the instance id. Except that it cannot be 0, because that means that the context is not valid.

Reply:
Ok, cleaned here https://github.com/micro-ROS/rcl/pull/16/commits/a5d197d0029b67d59998f60d82ab3775274c5a7f